src/main/java/controller/BlackJackController.java

```diff
@@ -0,0 +1,82 @@
+package controller;
+
+import domain.Card;
+import domain.Deck;
+import domain.Game;
+import domain.RandomShuffler;
+import java.util.List;
+import java.util.function.Supplier;
+import view.Answer;
+import view.InputView;
+import view.OutputView;
+
+public class BlackJackController {
+    private final InputView inputView = new InputView();
+    private final OutputView outputView = new OutputView();
+
+    public void run() {
+        Game game = retryUntilSuccess(this::startGame);
+        outputView.displayInitialDeal(game);
+        giveAdditionalCardsForPlayer(game);
+        giveAdditionalCardsForDealer(game);
+        displayScores(game);
+        displayGameResult(game);
+    }
+
+    public Game startGame() {
+        List<String> playerNames = inputView.readPlayerNames();
+        return new Game(playerNames, new Deck(new RandomShuffler()));
+    }
+
+    private void giveAdditionalCardsForPlayer(Game game) {
+        game.getPlayerNames().forEach((name) -> hitOrStay(game, name));
+    }
+
+    private void giveAdditionalCardsForDealer(Game game) {
+        outputView.displayEmptyLine();
+        while (game.doesDealerNeedCard()) {
+            game.dealerHit();
+            outputView.displayDealerHitResult();
+        }
+        outputView.displayEmptyLine();
+    }
+
+    private void displayScores(Game game) {
+        outputView.displayScore(game);
+    }
+
+    private void displayGameResult(Game game) {
+        game.calculateGameResult();
+        outputView.displayGameResult(game);
+    }
+
+    private void hitOrStay(Game game, String playerName) {
+        Answer answer = Answer.YES;
+        while (game.canHit(playerName) && answer == Answer.YES) {
+            answer = retryUntilSuccess(() -> inputView.readHitOrStay(playerName));
+            hitByAnswer(game, playerName, answer);
+            displayPlayerCards(game, playerName);
+        }
+    }
```

Comment:
while 문 내부에서 대답을 받아보는것 어떠신가요! 😄 

```suggestion
    private void hitOrStay(Game game, String playerName) {
        Answer answer = Answer.YES;
        while (game.canHit(playerName) && retryUntilSuccess(() -> inputView.readHitOrStay(playerName)) == Answer.YES) {
            game.playerHit(playerName);
            displayPlayerCards(game, playerName);
        }
    }
```

Reply:
while문 내부에서 대답 받는 방식은 생각하지 못했네요! 👍👍

다만 미션 예시에서는 `Answer.NO` 인 경우에도 카드를 보여주고 있어서 `displayPlayerCards()`가 실행되어야 하는데 while문에서 대답을 받으면 실행되지 않게 됩니다 😭

> jason는 한장의 카드를 더 받겠습니까?(예는 y, 아니오는 n)
n
jason카드: 7클로버, K스페이드

Reply:
아래와 같은 코드는 어떻게 생각하시나요? 🤔 

```suggestion
    private void hitOrStay(Game game, String playerName) {
        while (game.canHit(playerName)) {
            Answer answer = retryUntilSuccess(() -> inputView.readHitOrStay(playerName));
            if (answer == Answer.NO) {
                displayPlayerCards(game, playerName);
                return;
            }
            game.playerHit(playerName);
            displayPlayerCards(game, playerName);
        }
    }
```